server/src/main/java/org/opensearch/index/IndexModule.java

```diff
@@ -407,7 +410,9 @@ public enum Type {
         MMAPFS("mmapfs"),
         SIMPLEFS("simplefs"),
         FS("fs"),
-        REMOTE_SNAPSHOT("remote_snapshot");
+        REMOTE_SNAPSHOT("remote_snapshot"),
+        REMOTE_SEARCH("remote_search"),
```

Comment:
I think the naming is confusing here. 
Does `remote_search` mean read only index using remote store and `remote_warm_index` mean write capable remote index?